lib/CMakeLists.txt

```diff
@@ -480,18 +480,9 @@ if(QUDA_QIO)
 endif()
 
 if(QUDA_QDPJIT)
-  target_compile_definitions(quda PUBLIC USE_QDPJIT)
-  target_include_directories(quda SYSTEM PUBLIC $<BUILD_INTERFACE:${QUDA_QDPJITHOME}/include>)
-  target_link_libraries(
-    quda
-    INTERFACE ${QDP_LDFLAGS}
-              ${QDP_LIB}
-              ${QDP_LIBS}
-              ${QIO_LIB}
-              ${LIME_LIB}
-              ${QUDA_QMP_LDFLAGS}
-              ${QMP_LIB}
-              MPI::MPI_CXX)
+   target_compile_definitions(quda PRIVATE USE_QDPJIT)
+   # this avoids issues when qdpjit is build as a static library we should eventually switch to target_link_libraries(quda PRIVATE QDPXX::qdp)	
+   target_include_directories(quda PRIVATE $<TARGET_PROPERTY:QDPXX::qdp,INTERFACE_INCLUDE_DIRECTORIES>)
```

Comment:
Why not just:

target_link_libraries(quda PUBLIC QDPXX::qdp)
?

Reply:
I used that initially but when build qdpjit as a static library cmake still passed that to the linker command for libquda.so and that will fail as mentioned in the comment.
Using this workaround is fine as we never link any quda tests with qdpjit enabled and the link step in Chroma gets everything resolved.
Not sure this is an issue cmake should be able to handle better or there is a better solution for this.

Reply:
We should not preclude being able to build the tests with qdpxx enabled.  Adding support for doing so is highly desirable for testing, and it’s something Balint and I discussed as getting fixed.

Reply:
We can fix that when we get there.

Reply:
If it's a simply case of adding `QDP_initialize()` and `QDP_finalize()` to the QUDA tests to enable the tests to work with `QUDA_QDPJIT` enabled, can we not add that to this PR, and resolve @bjoo's concern now?

Reply:
So ok I guess the issue is that we can't link the .a to a .so if QDP-JIT is built statically. That is a reasonable comment. However, it seems like this is a bit odd use of CMake, since now we in principle don't propagate linking transitively. The client needs to explicitly do it. That seems odd to me. Right now we are saved by the fact that the only QDP-JIT client is Chroma which over QDP-JIT links things anyway.  I guess we need to choose which ugliness we take: users mixing static/dynamic linking and failing, or not propagating dependency information in the CMake way to clients (of which there is only one and may be one forever...)

Reply:
Yes, this is not _how it should be_. Will look into fixing this but for now it allows us to compile recent Chroma.

Reply:
About the tests: I don't actually know what needs to be done here. I have created #1222